### ISSUE TITLE
Some tests fail under Clang in when BUILD_TYPE=Release

### ISSUE THREAD

**Issue by avalenzu** [flexible-collision-library/fcl#218]
Executing the following
```
export CC=path/to/clang
export CXX=path/to/clang++
mkdir build
cd build
cmake BUILD_TYPE=Release ..
make
ctest
```
Yields
```
91% tests passed, 2 tests failed out of 22

Total Test time (real) =  46.23 sec

The following tests FAILED:
          9 - test_fcl_collision (Failed)
         13 - test_fcl_geometric_shapes (Failed)
```
These appear to be due to a strange interaction between Clang and Eigen inside `box_box-inl.h`. The result is false negatives in box-box intersection tests (missed contacts). I'm investigating possible solutions. I notice that the Travis CI tests clang in Debug mode, but not in Release. Could we consider adding that?

**Comment by jslee02** [comment]
> I notice that the Travis CI tests clang in Debug mode, but not in Release. Could we consider adding that?

I believe [we test clang builds in Release mode as well](https://github.com/flexible-collision-library/fcl/blob/master/.travis.yml#L21). Hm, am I missing something here?

**Comment by avalenzu** [comment]
No, I just missed that. Maybe it's a difference in clang versions? The test
results I posted were with 3.9. I'm not sure what Travis uses.

On Jun 15, 2017 4:22 PM, "Jeongseok Lee" <notifications@github.com> wrote:

I notice that the Travis CI tests clang in Debug mode, but not in Release.
Could we consider adding that?

I believe we test clang builds in Release mode as well
<https://github.com/flexible-collision-library/fcl/blob/master/.travis.yml#L21>.
Hm, am I missing something here?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/flexible-collision-library/fcl/issues/218#issuecomment-308855182>,
or mute the thread

**Comment by jslee02** [comment]
FYI, [Travis uses 3.5.0](https://travis-ci.org/flexible-collision-library/fcl/jobs/243278093#L673).
